app/Models/Server.php

```diff
@@ -102,111 +104,116 @@ public function __construct()
     {
-        if (self::$user->id === $server->owner || self::$user->root_admin === 1) {
-            return $server->daemonSecret;
+        $query = self::with('service', 'node')->where(function ($q) use ($uuid) {
+            $q->where('uuidShort', $uuid)->orWhere('uuid', $uuid);
+        });
+
+        if (! Auth::user()->isRootAdmin()) {
+            $query->whereIn('id', Auth::user()->serverAccessArray());
         }
 
-        $subuser = Subuser::where('server_id', $server->id)->where('user_id', self::$user->id)->first();
+        $result = $query->first();
 
-        if (is_null($subuser)) {
-            return null;
+        if (! is_null($result)) {
+            $result->daemonSecret = Auth::user()->daemonToken($result);
         }
 
-        return $subuser->daemonSecret;
+        return $result;
     }
 
     /**
-     * Returns array of all servers owned by the logged in user.
-     * Returns all users servers if user is a root admin.
+     * Returns non-administrative headers for accessing a server on the daemon.
      *
-     * @return \Illuminate\Database\Eloquent\Collection
+     * @param  string $uuid
+     * @return array
      */
-    public static function getUserServers($paginate = null)
+    public function guzzleHeaders()
     {
-        $query = self::select(
-            'servers.*',
-            'nodes.name as nodeName',
-            'locations.short as a_locationShort',
-            'allocations.ip',
-            'allocations.ip_alias',
-            'allocations.port',
-            'services.name as a_serviceName',
-            'service_options.name as a_serviceOptionName'
-        )->join('nodes', 'servers.node', '=', 'nodes.id')
-        ->join('locations', 'nodes.location', '=', 'locations.id')
-        ->join('services', 'servers.service', '=', 'services.id')
-        ->join('service_options', 'servers.option', '=', 'service_options.id')
-        ->join('allocations', 'servers.allocation', '=', 'allocations.id');
-
-        if (self::$user->root_admin !== 1) {
-            $query->whereIn('servers.id', Subuser::accessServers());
-        }
-
-        if (is_numeric($paginate)) {
-            return $query->paginate($paginate);
-        }
-
-        return $query->get();
+        return [
+            'X-Access-Server' => $this->uuid,
+            'X-Access-Token' => Auth::user()->daemonToken($this),
+        ];
     }
 
     /**
-     * Returns a single server specified by UUID.
-     * DO NOT USE THIS TO MODIFY SERVER DETAILS OR SAVE THOSE DETAILS.
-     * YOU WILL OVERWRITE THE SECRET KEY AND BREAK THINGS.
+     * Return an instance of the Guzzle client for this specific server using defined access token.
      *
-     * @param  string $uuid The Short-UUID of the server to return an object about.
-     * @return \Illuminate\Database\Eloquent\Collection
+     * @return \GuzzleHttp\Client
      */
-    public static function getByUUID($uuid)
+    public function guzzleClient()
```

Comment:
I think this way works, as it returns the client which is then used. Don't think we necessarily need to make it more complex.